test/system/trader.js

```diff
@@ -272,19 +274,20 @@ class Trader {
     }
   }
 
-  getLastPriceAndQuantity(price, quantity, side) {
-    orderbook[this.config.name][side].prices.push(price);
-    orderbook[this.config.name][side].quantities.push(quantity);
+  getLastPriceAndQuantity(price, quantity) {
+    orderbook[this.config.name]['B'].prices.push(price);
+    orderbook[this.config.name]['S'].quantities.push(quantity);
   }
 
   createOrder(order, options ={}) {
-    order.market = this.market;
-
+    if (!order.market) {
+      order.market = this.market;
+    }
     return this.client.createOrder(order)
       .then(
         result => {
           if (options.expectFail) {
-            throw new Error('createOrder succeeded when expected to fail');
+            throw new Error('Create Order failed due to  ' + options.expectFailReason);
```

Comment:
Change this text to:

'Create Order succeeded but was expected to fail due to' + options.expectFailReason